core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -2226,7 +2226,11 @@ public static String getFilePathExternalFilePath(String path, Configuration conf
       if (dataFile.getName().endsWith(CarbonCommonConstants.FACT_FILE_EXT)) {
         return dataFile.getAbsolutePath();
       } else if (dataFile.isDirectory()) {
-        return getFilePathExternalFilePath(dataFile.getAbsolutePath(), configuration);
+        if (getFilePathExternalFilePath(dataFile.getAbsolutePath(), configuration) == null) {
```

Comment:
Please  add/modify existing test case to handle this scenario

Also please test query with an empty folder. we have other places in the code where we recursively read.
Review and handle if a problem exists in that flow.

example:
CarbonTable.getFirstIndexFile --> also has same issue

Reply:
Handled the review comment and added the test cases also for the scenarios.